src/components/Filters/Filters.js

```diff
@@ -7,37 +7,161 @@ import FilterOffender from './FilterOffender';
 import FilterLocation from './FilterLocation';
 import FilterOther from './FilterOther';
 import './Filters.css';
+import produce from "immer";
 
-export default function Filters(props) {
-    function reducer(_, action) {
-        switch (action.type) {
-            case 'selection':
-                return { active: action.filterName };
-            default:
-                throw new Error();
+const testData = {
+    agents: [
+        { value: 'Andrew', supervisor: 'Sarah' },
+        { value: 'Steve', supervisor: 'Susan' },
+        { value: 'Karl', supervisor: 'Sarah' },
+        { value: 'Matt', supervisor: 'Susan' },
+        { value: 'Mitt', supervisor: 'Sarah' },
+    ],
+    supervisors: [
+        { value: 'Sarah' },
+        { value: 'Susan' },
+    ]
+};
+
+const loggedInUser = 'logged in user';
+
+const vanityCheck = (agentList) => {
```

Comment:
It feels funny to me to have this logic be in Filters rather than in `FilterAgent`. Same with the logic within the reducer under the `UPDATE_AGENT_LIST`. It seems to go against the mantra that components should be independent, reusable pieces that can be thought about in isolation.

Reply:
Lifting state is super common, even pre hooks.

Reply:
I would argue the component is still a reusable piece even though it doesn't manage it's state. Think about redux where no state is managed in a component.